ioservice/storage_dev.c

```diff
@@ -292,7 +295,8 @@ static int storage_dev_update_by_conf(struct m0_storage_dev  *dev,
 
 	M0_PRE(m0_ref_read(&dev->isd_ref) == 1);
 	m0_storage_dev_detach(dev);
-	rc = m0_storage_dev_new_by_conf(storage_devs, sdev, false, &dev_new);
+	rc = m0_storage_dev_new_by_conf(storage_devs, sdev,
+				        group_nr, index_nr, false, &dev_new);
```

Comment:
Can you explain why this change is needed?

Reply:
m0_storage_dev_new_by_conf() call storage_dev_new().  storage_dev_new() now required device size, and extra parameter as number of groups per device and number of index(btree). Ideally this should be attribute in sdev. Now i am sending as parameter. in future, we can do above change.